### PR TITLE
Template: Modify devcontainer-template.json properties

### PR DESCRIPTION
**Pull request by samruddhikhandale**

### CODE REVIEW

File: proposals/devcontainer-templates.md [hunk 1]
```diff
@@ -47,22 +47,20 @@ The `options` property contains a map of option IDs and their related configurat
       "type": "string",
       "description": "Description of the option",
       "proposals": ["value1", "value2"],
-      "default": "value1",
-      "replaceIn": [".devcontainer.json"],
```
Comment (samruddhikhandale) [inline proposals/devcontainer-templates.md:51]: Removed `replaceIn` as the tooling can find and replace in all the files within the Template folder.

File: proposals/devcontainer-templates.md [hunk 2]
```diff
@@ -26,19 +26,19 @@ The properties of the file are as follows:
 | Property | Type | Description |
 | :--- | :--- | :--- |
 | `id` | string | ID of the Template. The `id` should be unique in the context of the repository/published package where the Template exists and must match the name of the directory where the `devcontainer-template.json` resides. |
+| `version` | string | The semantic version of the Template. |
 | `name` | string | Name of the Template. |
 | `description` | string | Description of the Template. |
 | `documentationURL` | string | Url that points to the documentation of the Template. |
 | `licenseURL` | string | Url that points to the license of the Template. |
-| `type` | string | Type of the dev container (image, dockerfile, dockerCompose) created by the Template. |
```
Comment (samruddhikhandale) [inline proposals/devcontainer-templates.md:33]: CLI template package/publish commands will add the logic to determine it's value.

Comment (samruddhikhandale) [inline proposals/devcontainer-templates.md:33]: @jkeech when `type=image`, we can not be sure if the dev container only consists of a single `.devcontainer.json` ([example](https://github.com/devcontainers/templates/tree/main/src/jekyll/.devcontainer)) which can be used by dotcom.

Do you recommend adding another property (say `isSingleConfigFile`: boolean) to the `devcontainer-template.json` (which will be automatically added/computed by the cli publish cmd) ?

Comment (jkeech) [reply proposals/devcontainer-templates.md:33]: I think the publish command should inject something like `fileCount: number` when it's processing a feature. It shouldn't need to be manually entered into devcontainer-template.json. Doing it automatically would also prevent the declared value from getting out of sync with the template contents.

**Review by chrmarti — APPROVED** [review]
LGTM!

**Review by fuller53 — APPROVED** [review]

**Pull request merged**